docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/cgdp-definition/__tests__/cgdp-definition-legacy-helpers.test.ts

```diff
@@ -0,0 +1,185 @@
+			const mockModal = {
+				handleModalOpen: jest.fn(),
+				updateDialog: jest.fn(),
+			} as unknown as USAModal;
+			addListenersToDrupalGlossaryTerms(mockModal);
+			const link = document.querySelector('a.definition') as HTMLElement;
+			expect(link).not.toHaveAttribute('onclick');
+
+			// Use fireEvent to simulate the click
+			fireEvent.click(link);
+			expect(mockModal.handleModalOpen).toHaveBeenCalled();
+		});
+		it('should not try to add event listeners if there is no glossary links', () => {
+			document.body.innerHTML = `<a class="definition" data-glossary-name="Cancer.gov" data-glossary-audience="patient"></a>`;
+			const mockModal = {
+				handleModalOpen: jest.fn(),
+				updateDialog: jest.fn(),
+			} as unknown as USAModal;
+			addListenersToDrupalGlossaryTerms(mockModal);
+			const link = document.querySelector('a.definition') as HTMLElement;
+
+			// Use fireEvent to simulate the click
+			fireEvent.click(link);
+			expect(mockModal.handleModalOpen).not.toHaveBeenCalled();
+		});
+	});
+
+	describe('addListenersToPdqAndLegacyGlossaryTerms', () => {
+		it('should add event listeners to legacy glossary links that DO NOT have a dictionary', () => {
+			document.body.innerHTML = `<a class="definition" href="/Common/PopUps/popDefinition.aspx?id=CDR000123&amp;version=Patient&amp;language=English"></a>`;
+			const mockModal = {
+				handleModalOpen: jest.fn(),
+				updateDialog: jest.fn(),
+			} as unknown as USAModal;
+			addListenersToPdqAndLegacyGlossaryTerms(mockModal);
+			const link = document.querySelector('a.definition');
+			expect(link).not.toHaveAttribute('onclick');
+
+			// Use fireEvent to simulate the click
+			fireEvent.click(link as Element);
+			expect(mockModal.handleModalOpen).toHaveBeenCalled();
+		});
+		it('should add event listeners to legacy glossary links that have a dictionary', () => {
+			document.body.innerHTML = `<a class="definition" href="/Common/PopUps/popDefinition.aspx?id=CDR000123&amp;version=Patient&amp;language=English&amp;dictionary=genetic"></a>`;
+			const mockModal = {
+				handleModalOpen: jest.fn(),
+				updateDialog: jest.fn(),
+			} as unknown as USAModal;
+			addListenersToPdqAndLegacyGlossaryTerms(mockModal);
+			const link = document.querySelector('a.definition') as HTMLElement;
+			expect(link).not.toHaveAttribute('onclick');
+
+			// Use fireEvent to simulate the click
+			fireEvent.click(link);
+			expect(mockModal.handleModalOpen).toHaveBeenCalled();
+		});
+		it('should not try to add event listeners if there is no glossary links', () => {
+			document.body.innerHTML = `<a href="/Common/PopUps/popDefinition.aspx?id=CDR000123&amp;version=Patient&amp;language=English"></a>`;
+			const mockModal = {
+				handleModalOpen: jest.fn(),
+				updateDialog: jest.fn(),
+			} as unknown as USAModal;
+			addListenersToPdqAndLegacyGlossaryTerms(mockModal);
```

Comment:
Can you add the test for `a[onclick*="javascript:popWindow('defbyid'"]`` use case?